common/thorhelper/thorread.cpp

```diff
@@ -1073,6 +1074,378 @@ size32_t CsvDiskRowReader::getFixedDiskRecordSize()
+    while (xmlParser->next())
+    {
+        if (lastMatch)
+        {
+            ((CFieldFetcher *)fieldFetcher.get())->setCurrentMatch(lastMatch);
+            size32_t resultSize = translator->translate(builder, *this, *fieldFetcher);
+            dbgassertex(resultSize);
+            lastMatch.clear();
+            const void *ret = builder.getSelf();
+
+            if (fieldFilterMatchProjected(ret))
+            {
+                builder.finishRow(resultSize);
+                return ret;
+            }
+            else
+                builder.removeBytes(resultSize);
+        }
+    }
+    return eofRow;
+}
+
+void MarkupDiskRowReader::stop()
+{
+    opened = false;
+}
+
+bool MarkupDiskRowReader::matches(const char * format, bool streamRemote, IDiskReadMapping * otherMapping)
+{
+    if (!strieq(format, "markup"))
+        return false;
+    return ExternalFormatDiskRowReader::matches(format, streamRemote, otherMapping);
+}
+
+// IDiskRowReader
+size32_t MarkupDiskRowReader::getFixedDiskRecordSize()
+{
+    size32_t fixedDiskRecordSize = actualDiskMeta->getFixedSize();
+    if (fixedDiskRecordSize && grouped)
+        fixedDiskRecordSize += 1;
+    return fixedDiskRecordSize;
+}
+
+bool MarkupDiskRowReader::checkOpen()
+{
+    class CSimpleStream : public CSimpleInterfaceOf<ISimpleReadStream>
+    {
+        Linked<ISerialStream> stream;
+    public:
+        CSimpleStream(ISerialStream *_stream) : stream(_stream)
+        {
+        }
+    // ISimpleReadStream impl.
+        virtual size32_t read(size32_t max_len, void * data) override
+        {
+            size32_t got;
+            const void *res = stream->peek(max_len, got);
+            if (got)
+            {
+                if (got>max_len)
+                    got = max_len;
+                memcpy(data, res, got);
+                stream->skip(got);
+            }
+            return got;
+        }
+    };
+
+    if (!opened)
+    {
+        Owned<ISimpleReadStream> simpleStream = new CSimpleStream(inputStream);
+        if (kind==TAKjsonread)
+            xmlParser.setown(createJSONParse(*simpleStream, xpath, *this, noRoot?ptr_noRoot:ptr_none, useXmlContents));
+        else
+            xmlParser.setown(createXMLParse(*simpleStream, xpath, *this, noRoot?ptr_noRoot:ptr_none, useXmlContents));
+
+        if (!fieldFetcher)
+            fieldFetcher.setown(new CFieldFetcher(*record, nullptr));
+
+        opened = true;
+        return true;
+    }
+    return false;
+}
+
+
+class XmlDiskRowReader : public MarkupDiskRowReader
+{
+public:
+    XmlDiskRowReader(IDiskReadMapping * _mapping)
+    : MarkupDiskRowReader(_mapping, TAKxmlread)
+    {
+        const IPropertyTree & fileOptions = *mapping->queryFileOptions();
+        const IPropertyTree & xmlOptions = *fileOptions.queryPropTree("formatOptions");
+
+        if (customRowTag.isEmpty()) // no override
+            xmlOptions.getProp("xpath", xpath);
+        else
+        {
+            xpath.set("/Dataset/");
```

Comment:
Please open a jira and release a PR to apply this fix to the code in dafsserver.cpp

Reply:
Here is the JIRA for that fix: https://track.hpccsystems.com/browse/HPCC-31424